libraries/provision/ansible/playbooks/tasks/create-sync-gateway-user.yml

```diff
@@ -1,7 +1,46 @@
-# Add sync_gateway user
-- name: SYNC GATEWAY | Create sync gateway user
-  become: yes
+# Add sync_gateway user on Centos/linux/ubuntu/debian
+- name: SYNC GATEWAY | Create sync gateway user on Centos/linux/ubuntu/debian
   user: name=sync_gateway createhome=yes
+  when: ansible_distribution == "CentOS" or ansible_distribution == "RedHat" or ansible_distribution == "Ubuntu" or ansible_distribution == "Debian"
+
+# # Add sync_gateway user on macOS 
+#- name: SYNC GATEWAY | Create sync gateway user on MacOS
+#  become: yes
+#  shell: |
+#    sysadminctl -addUser sync_gateway
+#    dseditgroup -o create sync_gateway
+#    dseditgroup -o edit -a sync_gateway -t user sync_gateway
+#  when: ansible_distribution == "MacOSX"
+
+# # Add sync_gateway user on macOS 
+# - name: SYNC GATEWAY | Add sync gateway user on MacOS
+#   become: yes
+#   shell: sysadminctl -addUser sync_gateway -adminUser sridevisaragadam -adminPassword TasRhs_1708
+#   when: ansible_distribution == "MacOSX"
+
+# - name: SYNC GATEWAY | Create group policy for sync gateway on MacOS
+#   become: yes
+#   shell: dseditgroup -o create sync_gateway
+#   when: ansible_distribution == "MacOSX"
+
+# - name: SYNC GATEWAY | Add Sync gateway user to sync gateway group on MacOS
+#   become: yes
+#   shell: dseditgroup -o edit -a sync_gateway -t user sync_gateway
+#   when: ansible_distribution == "MacOSX"
```

Comment:
may these code still have a chance to be rollback for debugging convenience? if not, it's better to get rid of them.

Reply:
As we discussed , I am leaving commented and added comment with reason why this is left as comment.